app/code/Magento/Catalog/Model/Product.php

```diff
@@ -668,7 +668,7 @@ public function getTypeId()
      */
     public function getStatus()
     {
-        $status = $this->_getData(self::STATUS);
+        $status = (int) $this->_getData(self::STATUS);
         return $status !== null ? $status : Status::STATUS_ENABLED;
```

Comment:
The following ternary is now useless, as the cast on line 671 means that the value can now never be `null`.
```suggestion
        $status = $this->_getData(self::STATUS);
        return $status !== null ? (int) $status : Status::STATUS_ENABLED;
```